src/TLuaInterpreter.cpp

```diff
@@ -17158,49 +17162,115 @@ void TLuaInterpreter::initIndenterGlobals()
 
-
-
 #if defined(Q_OS_MACOS)
-        //macOS app bundle would like the search path to also be set to the current binary directory
-        luaL_dostring(pIndenterState.get(), QStringLiteral("package.cpath = package.cpath .. ';%1/?.so'")
-                      .arg(QCoreApplication::applicationDirPath())
-                      .toUtf8().constData());
-        luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = package.path .. ';%1/?.lua'")
-                      .arg(QCoreApplication::applicationDirPath())
-                      .toUtf8().constData());
+    // macOS app bundle would like the search path for the binary modules to
+    // also be set to the current binary directory:
+    luaL_dostring(pIndenterState.get(), QStringLiteral("package.cpath = '%1/?.so;' .. package.cpath")
+                  .arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+#endif
 
-#elif defined(Q_OS_UNIX)
-    // Need to tweak the lua path for the installed *nix case and AppImage builds as well as
-    // to allow running from a shadow build directory (both qmake and cmake).
-    luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = '" LUA_DEFAULT_PATH "/?.lua;%1/?.lua;%1/../3rdparty/?.lua;%1/../../3rdparty/?.lua;%1/../../mudlet/3rdparty/?.lua;' .. package.path")
-                  .arg(QCoreApplication::applicationDirPath())
-                  .toUtf8().constData());
+#if defined(Q_OS_UNIX)
+    // Now also included for the MacOs case so that MacOS developers can build
+    // from within the Qt Creator IDE
+
+    /*
+     * Additional paths for the lua package search path - ? is the "package"
+     * name though in the case for the lua code formatter it itself includes
+     * leading path elements of "lcf" and "workshop" in the
+     * require("lcf.workshop.base") case - the lua script being sought is the
+     * "base.lua" file but our paths must give the loction of the "lcf"
+     * directory two parent directories up:
+     * 1 installed *nix case - probably not applicable to Windows
+     *     "LUA_DEFAULT_PATH/?.lua" (if defined and not empty)
+     * 2 AppImage (directory of executable):
+     *     "<applicationDirectory>/?.lua"
+     * 3 QMake shadow builds without CONFIG containing "debug_and_release" but
+     *   with "debug_and_release_target" (default on most OS but NOT Windows):
+     *     "<applicationDirectory>/../3rdparty/?.lua"
+     * 4 QMake shadow builds with CONFIG containing "debug_and_release" AND
+     *   "debug_and_release_target" (usually Windows):
+     *     "<applicationDirectory>/../../3rdparty/?.lua"
+     * 5 CMake shadow builds
+     *     "<applicationDirectory>/../../mudlet/3rdparty/?.lua"
+     * NOTE: In the cmake case the shadow build directory is alongside the base
+     * directory of the extracted source package so it MUST be extracted into
+     * a known directory name which is assumed here to be "mudlet"!
+     */
```

Comment:
This is a case of too much information, I think. We don't need to explain how lua require() works in our C++ files complete with example. This beast of a file is already 17k+ lines long, we don't need to make it even harder for our editors to open. 

As for delineating each case, I can see an argument for this but I feel like the right place to do it is above each entry when you add it. Which will be made easier if you take my suggestion to keep the code DRY below.

Reply:
I had it about each line in each case - twice (one for \*nix like cases and once for Windows) - and Vadim said I should cut it down. As for explaining each entry it took a far bit of research/hackery to work out why each line was needed and I don't want that information lost. For instance for CMake *shadow (out-of-source, the default these days)* builds it is ***essential*** to extract the mudlet source into a directory called, not surprisingly, "mudlet" - but that is not obvious!

Reply:
The middle of a 17,000 long file is definitely not the right place to have this information.

Reply:
Also:
>  We don't need to explain how lua require() works in our C++ files complete with example.

We need to explain it somewhere - it took me a while to workout that the '.'s in that `require` had significance and if I had had that explanation when I first tried porting Mudlet to both MSYS2 AND FreeBSD it would have made things go a lot smoother and without the head scratching around: WTF is `base.lua` and why is it that that file is being sought?! Given that this is the code that is looking for it it seems a perfectly logical place to say what it is looking for.

Reply:
It's explained in the Lua manual - part of the learning process. No Mudlet dev should have personal comments in the codebase, it's shared by all!

Reply:
There is nothing about that that is a personal comment? Where are you forming that opinion from? :confused:

Reply:
Well - I don't think others find it helpful, certainly me and demonnic don't. I think you'd be the only one to benefit from it. 

If there's already a proper manual explaining something, we shouldn't duplicate it inside Mudlet.

Reply:
This needs to go as well, this explanation is not necessary. If you'd like to record for yourself how the loader works, it's best to do it elsewhere - because forcing everyone to read 6 line explanation of how loading of just one particular module we use isn't nice to the team. And it's forced because comments are meant to be read - if we start ignoring comments by habit because they're long and not useful, that will be a bad situation!

Reply:
Some explanation *is* necessary IMHO!

I have shorted it to three lines and included a link to a very informative SE QA - some of the details there (about '.'s being used as directory separators - do not *even* get mentioned in the PIL 5.1 manual) - which is why I felt it necessary to put something in for those of us who are **NOT** Lua gurus...!

Reply:
I still don't think it's helpful. You'll be looking at the Lua manual anyway to learn it. There is no added value by putting it here, except making everyone read it, and not everyone wants to read this. :confused: